.circleci/config.yml

```diff
@@ -0,0 +1,126 @@
+            fi
+
+            export PATH="$HOME/.cabal/bin:$HOME/.ghcup/bin:$PATH"
+
+            ghcup set "8.6.3" || ghcup install "8.6.3" && ghcup set "8.6.3"
+            command -v cabal || ghcup install-cabal "2.4.1.0"
+
+            if ! [ -x $HOME/.local/bin/stack ]; then
+              curl -sSL https://get.haskellstack.org/ | sh
+            fi
+
+            stack config set system-ghc --global true
+            stack config set install-ghc --global false
+
+            echo 'export PATH="$HOME/.cabal/bin:$HOME/.ghcup/bin:$PATH"' >> $BASH_ENV
+
+      - save_cache:
+          key: toolchain-{{ arch }}-{{ epoch }}
+          paths:
+            - "~/.ghcup"
+
+      - run:
+          name: Homebrew Dependencies
+          command: |
+            brew install coreutils
+            if [[ "$CIRCLE_BRANCH" == "master" ]]; then
+              brew tap caskroom/cask
+              brew cask install google-cloud-sdk
+            fi
+
+      - save_cache:
+          key: homebrew-{{ epoch }}
+          paths:
+            - /usr/local/Homebrew
+
+      - run:
+          name: Build Dependencies
+          command: stack build --no-terminal --dependencies-only
+
+      - save_cache:
+          key: stack-root-{{ arch }}-ghc8.6.3-{{ checksum "snapshot.yaml" }}-{{ checksum "package.yaml" }}
+          paths:
+            - "~/.stack"
+
+      - run:
+          name: Build Executables
+          command: |
+            cabal v1-update
+            cabal v1-configure \
+              --prefix=/usr/local \
+              --docdir=/usr/local/share/doc/radicle \
+              --libsubdir=radicle \
+              --datasubdir=radicle \
+              --libexecsubdir=radicle \
+              --package-db="$(stack path --snapshot-pkg-db)" \
+              --package-db="$(stack path --local-pkg-db)" \
+              --package-db="$(stack path --global-pkg-db)"
+            cabal v1-build
+
+      - run:
+          name: Package
+          command: |
+            if [[ "$CIRCLE_BRANCH" == "master" ]]; then
+              cabal v1-copy \
+                exe:rad \
+                exe:radicle \
+                exe:radicle-daemon \
+                exe:rad-machines \
+                --destdir=dist
+
+              mkdir -p dist/pkg
+              cd dist/pkg
+
+              tarball=radicle_${CIRCLE_SHA1}_x86_64-darwin.tar.gz
+              tar -C .. -cvzf $tarball usr
+              sha512sum $tarball|cut -d ' ' -f1 > ${tarball}.sha512
+            fi
+
+      - run:
+          name: Upload to Google Cloud Storage
+          command: |
+            if [[ "$CIRCLE_BRANCH" == "master" ]]; then
+              # Key ID: db4e4a0b0ee696ea6f263c20de593c2fba4d0998
+              echo "$GCLOUD_SERVICE_KEY" | \
+                gcloud auth activate-service-account \
+                  circleci-image-uploader@opensourcecoin.iam.gserviceaccount.com \
+                  --key-file=-
+              gcloud --quiet config set project ${GCLOUD_PROJECT_ID}
```

Comment:
Do we need this? If so could we hardcode the project ID or add it to the YAML file? Right now it is a secret variable which can only be discovered through the CircleCI UI and its value is hidden.